lib/matplotlib/tests/test_colors.py

```diff
@@ -44,6 +45,32 @@ def test_resample():
     assert_array_almost_equal(lc3([0, 0.5, 1]), expected)
 
 
+def test_colormap_copy():
+    cm = plt.cm.Reds
+    cm([-1, 0, .5, np.nan, np.inf])
+    cm.set_bad('y')
```

Comment:
@tacaswell Won't this leak into the global state of `plt.cm.Reds`? Why do we have global state anyway? Maybe the colormaps should be made read-only until copied?